### PR TITLE
Modify Univariate methods so that they don't store the entire reference data in the fit method 

### PR DESCRIPTION
**Pull request by Dbhasin1**
KS:

- [x] Implementation 
- [x] Docs 

Chi-Squared:

- [x] Implementation 

Wasserstein:

- [x] Implementation
- [x] Docs

### CODE REVIEW

File: nannyml/base.py [hunk 1]
```diff
@@ -177,6 +177,7 @@ def __init__(
         chunk_period: Optional[str] = None,
         chunker: Optional[Chunker] = None,
         timestamp_column_name: Optional[str] = None,
+        calculation_method: Optional[str] = None,
```
Comment (nnansters) [inline nannyml/base.py:180]: I would not go so far as to add this on the base class. If I understand it correctly the `calculation_method` parameter only applies to the univariate drift calculator, so I would only introduce it there.

File: nannyml/drift/univariate/calculator.py [hunk 2]
```diff
@@ -76,7 +80,12 @@ def __init__(
         ...    res.plot(kind='drift', column_name=column_name, method=method).show()
         """
         super(UnivariateDriftCalculator, self).__init__(
-            chunk_size, chunk_number, chunk_period, chunker, timestamp_column_name
+            chunk_size,
+            chunk_number,
+            chunk_period,
+            chunker,
+            timestamp_column_name,
+            calculation_method,
```
Comment (nnansters) [inline nannyml/drift/univariate/calculator.py:88]: Cfr. my comment on the base class: don't have the `calculation` in the `super` call here, but just add it as an instance variable here.

Comment (Dbhasin1) [reply nannyml/drift/univariate/calculator.py]: resolved, please take a look @nnansters

Comment (nnansters) [reply nannyml/drift/univariate/calculator.py]: Looks good!

**Pull request merged**